frontend/src/components/Workspace/Visualize/Plot/ImagePlot.tsx

```diff
@@ -476,66 +560,79 @@ const ImagePlotChart = memo(function ImagePlotChart({
-    const xAdd = Number(((sizeDrag.width + 2) / (sChart / sizeX)).toFixed(1))
-    const yAdd = Number(((sizeDrag.height + 2) / (sChart / sizeY)).toFixed(1))
-    const x = Number((sizeDrag.left / (sChart / sizeX)).toFixed(1))
-    const y = Number((sizeDrag.top / (sChart / sizeY)).toFixed(1))
-
-    const pointCenter = {
-      posx: x + Math.floor(xAdd / 2),
-      posy: y + Math.floor(yAdd / 2),
-      sizex: xAdd,
-      sizey: yAdd,
-    }
-    dispatch(resetAllOrderList())
-    try {
+    if (action === "Add ROI") {
+      setLoadingApi(true)
+      const sizeX = roiDataState[0].length - 1
+      const sizeY = roiDataState.length - 1
+      const xAdd = Number(((sizeDrag.width + 2) / (sChart / sizeX)).toFixed(1))
+      const yAdd = Number(((sizeDrag.height + 2) / (sChart / sizeY)).toFixed(1))
+      const x = Number((sizeDrag.left / (sChart / sizeX)).toFixed(1))
+      const y = Number((sizeDrag.top / (sChart / sizeY)).toFixed(1))
+
+      const pointCenter = {
+        posx: x + Math.floor(xAdd / 2),
+        posy: y + Math.floor(yAdd / 2),
+        sizex: xAdd,
+        sizey: yAdd,
+      }
+      dispatch(resetAllOrderList())
       await addRoiApi(roiFilePath, pointCenter)
-    } catch {
-      // TODO: add error message
+      setLoadingApi(false)
+      onCancelAdd()
+      workspaceId && dispatch(getRoiData({ path: roiFilePath, workspaceId }))
     }
-    setLoadingApi(false)
-    onCancelAdd()
-    workspaceId && dispatch(getRoiData({ path: roiFilePath, workspaceId }))
-    resetTimeSeries()
-  }
-
-  const onMergeRoi = async () => {
-    if (!roiFilePath || loadingApi) return
-    setLoadingApi(true)
-    dispatch(resetAllOrderList())
-    try {
+    if (action === "Merge ROI") {
+      if (pointClick.length < 2) return
+      setLoadingApi(true)
+      dispatch(resetAllOrderList())
       await mergeRoiApi(roiFilePath, {
         ids: pointClick.map((point) => point.z),
       })
-    } catch {
-      // TODO: add error message
+      setLoadingApi(false)
+      setPointClick([])
+      workspaceId && dispatch(getRoiData({ path: roiFilePath, workspaceId }))
+    } else if (action === "Delete ROI") {
+      if (!pointClick.length) return
+      setLoadingApi(true)
+      dispatch(resetAllOrderList())
+      await deleteRoiApi(roiFilePath, {
+        ids: pointClick.map((point) => point.z),
+      })
+      setLoadingApi(false)
+      setPointClick([])
+      workspaceId && dispatch(getRoiData({ path: roiFilePath, workspaceId }))
     }
-    setLoadingApi(false)
-    onCancel()
-    workspaceId && dispatch(getRoiData({ path: roiFilePath, workspaceId }))
     resetTimeSeries()
+    setEdit(true)
+  }
+
+  const onMergeRoi = async () => {
+    if (!roiFilePath || loadingApi) return
+    setAction("Merge ROI")
+    setEdit(false)
   }
 
   const onDeleteRoi = async () => {
     if (!roiFilePath || loadingApi) return
+    setEdit(false)
+    setAction("Delete ROI")
+  }
+
+  const onCommitRoi = async () => {
```

Comment:
@sanglevinh 
After ROI EDIT commit is completed, a completion message should be displayed.
→ UI assumes a snackbar similar to Workflow completion, etc.
→ Message is "Edit ROI Finished"

Reply:
@itutu-tienday i added it, pls check it again